app/controllers/api/v1/messages_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Api
+  module V1
+    class MessagesController < Api::V1::ApiController
+      def index
+        @messages = policy_scope(conversation.messages).includes(:user)
+      end
+
+      def create
+        @conversation = find_conversation
```

Comment:
I see you have some duplicated code, maybe you can remove `def find_conversation` method and use `conversation` method here
```suggestion
        @conversation = conversation
```

Reply:
Done!